src/app/public/modules/i18n/resources.service.ts

```diff
@@ -67,12 +69,25 @@ export class SkyAppResourcesService {
   /**
    * Gets a resource string based on its name.
    * @param name The name of the resource string.
+   * @param args Any templated args.
    */
   public getString(name: string, ...args: any[]): Observable<string> {
-    if (!this.resourcesObs) {
-      const localeObs = this.localeProvider.getLocaleInfo();
+    const localeObs: Observable<SkyAppLocaleInfo> = this.localeProvider.getLocaleInfo();
+    return this.getStringForLocaleInfoObservable(localeObs, name, ...args);
+  }
+
+  /**
+   * Gets a resource string for a specific locale based on its name.
+   * @param localeInfo The locale to use.
+   * @param name The name of the resource string.
+   * @param args Any templated args.
+   */
+  public getStringForLocale(localeInfo: SkyAppLocaleInfo, name: string, ...args: any[]): Observable<string> {
+    return this.getStringForLocaleInfoObservable(of(localeInfo), name, ...args);
+  }
 
-      this.resourcesObs = localeObs
+  private getStringForLocaleInfoObservable(localeInfoObs: Observable<SkyAppLocaleInfo>, name: string, ...args: any[]): Observable<string> {
+      const resourcesObs: Observable<any> = localeInfoObs
```

Comment:
I'd also like to try to keep the `resourceObs` cache in place. We probably just need to key the cache off the locale string from `localeInfo`, similar to how the `httpObs` is cached.

Reply:
Resource caching by locale added.